backends/memcheck/ceed-memcheck-restriction.c

```diff
@@ -60,7 +60,7 @@ static inline int CeedElemRestrictionApplyStridedNoTranspose_Memcheck_Core(CeedE
     CeedPragmaSIMD for (CeedInt k = 0; k < num_comp; k++) {
       CeedPragmaSIMD for (CeedInt n = 0; n < elem_size; n++) {
         CeedPragmaSIMD for (CeedInt j = 0; j < block_size; j++) {
-          vv[e * elem_size * num_comp + (k * elem_size + n) * block_size + j - v_offset] =
+          vv[e * elem_size * (CeedSize)num_comp + (k * (CeedSize)elem_size + n) * block_size + j - v_offset] =
```

Comment:
Note that the product is evaluated from left to right, with promotion only when required. So for example, this is UB (signed integer overflow), producing a negative result.
```c
    int a = 1024*1024, b = 2048, c = 3;
    long prod = a * b * (long)c;
```
https://godbolt.org/z/h8MveE9b3

I think we should either check that there cannot be overflow or promote the loop indices.

Reply:
I swapped the indices to be `CeedSize` 33e3c8898fbd5307119640c2a4eb4dede2c8b0eb

Reply:
I think this was addressed in [33e3c88](https://github.com/CEED/libCEED/pull/1528/commits/33e3c8898fbd5307119640c2a4eb4dede2c8b0eb), but @jeremylt can correct me if I'm wrong.

Reply:
Oh, good. I was looking at the wrong diff.